import-automation/executor/schedule_update_import.py

```diff
@@ -0,0 +1,234 @@
+            bucket, and then parsed and passed here as a configs.ExecutorConfig
+            object.
+        absolute_import_path: a string specifying the import's path in the
+            following format:
+                <path_to_directory_relative_to_repository_root>:<import_name>
+            example:
+                scripts/us_usda/quickstats:UsdaAgSurvey
+        import_script_args: a list of strings, each to be used as a command
+            line arg for the import script,
+            e.g. ['--flag1=value1', '--flag2=value2'].
+        local_repo_dir: the full path to the GitHub repository on local. The
+            path shoud be provided to the root  directory of the repo,
+            e.g. `<base_path_on_disk>/data`.
+
+    Returns:
+        An import_executor.ExecutionResult object.
+    """
+    # Update the configs with user script args, if provided.
+    if import_script_args:
+        cfg.user_script_args = import_script_args
+
+    executor = import_executor.ImportExecutor(
+        uploader=file_uploader.GCSFileUploader(
+            project_id=cfg.gcs_project_id,
+            bucket_name=cfg.storage_prod_bucket_name),
+        github=github_api.GitHubRepoAPI(
+            repo_owner_username=cfg.github_repo_owner_username,
+            repo_name=cfg.github_repo_name,
+            auth_username=cfg.github_auth_username,
+            auth_access_token=cfg.github_auth_access_token),
+        config=cfg,
+        local_repo_dir=local_repo_dir)
+
+    return executor.execute_imports_on_update(absolute_import_path)
+
+
+def main(_):
+    mode = _FLAGS.mode
+    absolute_import_path = _FLAGS.absolute_import_path
+    import_script_args = _FLAGS.import_script_args
+
+    if not _FLAGS.config_project_id:
+        raise Exception("Flag: config_project_if must be provided.")
+
+    if not mode or (mode not in ['update', 'schedule']):
+        raise Exception('Flag: mode must be set to \'update\' or \'schedule\'')
+
+    if not import_target.is_absolute_import_name(absolute_import_path):
+        raise Exception(
+            'Flag: absolute_import_path is invalid. Path should be like:'
+            'scripts/us_usda/quickstats:UsdaAgSurvey')
+
+    # Converting string to list
+    args_list = import_script_args.split(' ')
+    if type(args_list) != type([]):
+        raise Exception(
+            'Flag: import_script_args could not be parsed into a list.')
+
+    # Get the root repo directory (data). Assumption is that this script is being
+    # called from a path within the data repo.
+    cwd = os.getcwd()
+    repo_dir = cwd.split("data")[0] + "data"
+    logging.info(f'{mode} called with the following:')
+    logging.info(f'Config Project ID: {_FLAGS.config_project_id}')
+    logging.info(f'Import: {absolute_import_path}')
+    logging.info(f'Import script args: {args_list}')
+    logging.info(f'Repo root directory: {repo_dir}')
+
+    cfg = _get_cloud_config()
```

Comment:
Would it be useful to test with a local config file to try other config options like script timeout, user script args before updating config for a job? Should such configs always be uploaded to GCS to test?

Reply:
I can add that as a TODO where the user could provide a local config file to overwrite or add config params locally. Hopefully ok to do in a follow-up PR? 

Note that you are expected to provide the user script args locally as flag inputs to this script. So that won't be coming from GCS. However, the various auth params should only be hosted on GCS and I don't suspect those will change.

Reply:
Added the TODO.

Reply:
And here is the follow-up PR for this: https://github.com/datacommonsorg/data/pull/978

Reply:
As a result of this, I have also removed the option to provide user script args as a command line flag. Instead, they can be set in the local config json file in the follow-up PR. Thanks for the suggestion!